lms/djangoapps/grades/tasks.py

```diff
@@ -138,7 +146,7 @@ def _has_database_updated_with_new_score(
         if api_score is None:
             # Same case as the initial 'if' above, for submissions-specific scores
             return score_deleted
-        reported_modified_time = api_score.created_at
+        reported_modified_time = api_score['created_at']
```

Comment:
The change to this line will fix the issue noted on [TNL-6254](https://openedx.atlassian.net/browse/TNL-6245), as shown from my splunk discovery and subsequent debugging:

```
1/9/17
4:07:07.000 PM	
Jan  9 16:07:07 ip-10-2-70-117 [service_variant=lms][celery.worker.job][env:prod-edx-edxapp] ERROR [ip-10-2-70-117  1042] [log.py:282] - Task lms.djangoapps.grades.tasks.recalculate_subsection_grade_v2[1020a13f-87f8-43a3-9e54-c01cf2f6586c] raised unexpected: AttributeError("'ReturnDict' object has no attribute 'created_at'",)
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.74.0.54/newrelic/hooks/application_celery.py", line 84, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 96, in recalculate_subsection_grade_v2
    kwargs['user_id'], scored_block_usage_key, expected_modified_time, score_deleted,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 141, in _has_database_updated_with_new_score
    reported_modified_time = api_score.created_at
AttributeError: 'ReturnDict' object has no attribute 'created_at'
host = ip-10-2-70-117 source = /edx/var/log/lms/edx.log sourcetype = edx
```

```
> /edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py(142)_has_database_updated_with_new_score()
-> reported_modified_time = api_score.created_at
(Pdb) api_score
{'submission': 3L, 'created_at': datetime.datetime(2017, 1, 9, 16, 23, 8, 696028, tzinfo=<UTC>), 'points_earned': 3, 'submission_uuid': u'd952344a-d687-11e6-9cd7-080027880ca6', 'annotations': [], 'student_item': 2L, 'points_possible': 4}
(Pdb) api_score.created_at
*** AttributeError: 'ReturnDict' object has no attribute 'created_at'
(Pdb) api_score['created_at']
datetime.datetime(2017, 1, 9, 16, 23, 8, 696028, tzinfo=<UTC>)
```

It's also worth noting that this bug would have been discovered much more quickly had this PR's main work of expanding our exception logging in the grades update task been added earlier.